src/App.tsx

```diff
@@ -1,27 +1,53 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
+
+
+let a = Array.from({length:5}, (_, index)=>index)
+console.log(a);
+
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [select, setSelect] = useState('5');
+  const [currentPage, setCurrentPage] = useState(1);
+
+const handlePageChange = (page: number) => {
+  setCurrentPage(page);
+};
+
+function prepareList() {
+  const startIndex = (currentPage - 1) * parseInt(select);
+  const endIndex = startIndex + parseInt(select);
+  return items.slice(startIndex, endIndex);
+}
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {currentPage} (items {(currentPage - 1) * parseInt(select) + 1} - {Math.min(currentPage * parseInt(select), items.length)} of 42)
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
             id="perPageSelector"
-            className="form-control">
+            className="form-control"
+            value={select}
+            onChange={e => {
+              setSelect(e.target.value)
```

Comment:
Consider adding validation to ensure that `e.target.value` is a valid number before setting it as the new `select` value. This will prevent potential issues if an invalid value is somehow set.